torch/_tensor.py

```diff
@@ -149,6 +153,27 @@ def __reduce_ex__(self, proto):
                 state = self.__dict__
         return (_rebuild_from_type_v2, (func, type(self), args, state))
 
+    def storage(self):
+        r"""
+        storage() -> torch.Storage
+
+        Returns the underlying storage.
+        """
+        if has_torch_function_unary(self):
+            return handle_torch_function(Tensor.storage, (self,), self)
+
+        if self.dtype not in torch.storage._dtype_to_storage_type_map():
+            raise RuntimeError(f'unsupported Storage type: {self.dtype}')
```

Comment:
We should describe here what the eventual forward move here is, because one goal is to prevent people from having to keep defining cartesian product of device type x dtype storages whenever they add a new device type / dtype storage. It's not entirely clear what the best answer here is. One janky solution is to not bother with wrapping the storage type for anything that isn't in a BC list. I guess mid term it would be best not to return FloatStorage anymore (and just return a TypedStorage), it's been a while so I don't remember what we discussed here

Reply:
I don't remember talking about that. But yeah, we can change it to return TypedStorage, I don't see anything wrong with that idea. Should I do that now, or in a different PR? I don't think that should take long to change, but you never know

Reply:
The main reason to do it in this PR is if changing it to return a TypedStorage would be yet another BC breaking change, but I'm not sure that would actually be the case. If we're very close to all tests passing we can defer.